library/ssl_msg.c

```diff
@@ -5551,6 +5551,23 @@ void mbedtls_ssl_transform_free( mbedtls_ssl_transform *transform )
     mbedtls_platform_zeroize( transform, sizeof( mbedtls_ssl_transform ) );
 }
 
+void mbedtls_ssl_set_inbound_transform( mbedtls_ssl_context *ssl,
+                                        mbedtls_ssl_transform *transform )
+{
+    if( ssl->transform_in == transform )
+        return;
+
+    ssl->transform_in = transform;
+    mbedtls_platform_zeroize( ssl->in_ctr, 8 );
```

Comment:
Not having a macro for this 8 is not good practice. Could we take the time to improve that ?

Reply:
Only found two place can be replaced with macro. 

I did not understand other places. Below is a list .
```
library/ssl_msg.c:2992:    original_in_ctr = ssl->in_ctr;
library/ssl_msg.c:2995:    ssl->in_ctr = record_in_ctr;
library/ssl_msg.c:3000:    ssl->in_ctr = original_in_ctr;
library/ssl_msg.c:3010:    uint64_t rec_seqnum = ssl_load_six_bytes( ssl->in_ctr + 2 );
library/ssl_msg.c:3035:    uint64_t rec_seqnum = ssl_load_six_bytes( ssl->in_ctr + 2 );
library/ssl_msg.c:3444:        memcpy( &rec->ctr[0], ssl->in_ctr, rec_hdr_ctr_len );
library/ssl_msg.c:3533:    unsigned int rec_epoch = ( ssl->in_ctr[0] << 8 ) | ssl->in_ctr[1];
library/ssl_msg.c:3655:                if( ++ssl->in_ctr[i - 1] != 0 )
library/ssl_msg.c:4796:    mbedtls_platform_zeroize( ssl->in_ctr, MBEDTLS_SSL_IN_CTR_LEN );
library/ssl_msg.c:4884:        ssl->in_ctr = ssl->in_hdr +  3;
library/ssl_msg.c:4886:        ssl->in_cid = ssl->in_ctr +  8;
library/ssl_msg.c:4889:        ssl->in_len = ssl->in_ctr + 8;
library/ssl_msg.c:4896:        ssl->in_ctr = ssl->in_hdr - 8;
library/ssl_msg.c:5069:    in_ctr_cmp = memcmp( ssl->in_ctr + ep_len,
library/ssl_msg.c:5563:    mbedtls_platform_zeroize( ssl->in_ctr, MBEDTLS_SSL_IN_CTR_LEN );
library/ssl_srv.c:1217:        if( ssl->in_ctr[0] != 0 || ssl->in_ctr[1] != 0 )
library/ssl_srv.c:1223:        memcpy( ssl->cur_out_ctr + 2, ssl->in_ctr + 2, 6 );
library/ssl_tls.c:3260:    ssl->in_ctr = NULL;
```

Reply:
I'd suggest the following changes:
library/ssl_msg.c:3654:            for( i = MBEDTLS_SSL_IN_CTR_LEN; i > mbedtls_ssl_ep_len( ssl ); i-- )
library/ssl_msg.c:4886:        ssl->in_cid = ssl->in_ctr +  MBEDTLS_SSL_IN_CTR_LEN;
library/ssl_msg.c:4889:        ssl->in_len = ssl->in_ctr + MBEDTLS_SSL_IN_CTR_LEN;
library/ssl_msg.c:4896:        ssl->in_ctr = ssl->in_hdr - MBEDTLS_SSL_IN_CTR_LEN;
library/ssl_srv.c:1223:        memcpy( ssl->cur_out_ctr + 2, ssl->in_ctr + 2, MBEDTLS_SSL_IN_CTR_LEN - 2 );

Reply:
Mark,  not addressed this point